### ISSUE TITLE
put is not supported

### ISSUE THREAD

**Issue by siaron** [api7/etcd-adapter#13]
- Which version of apisix is currently supported?
- start Error when starting apisix 
```
xxx:apisix-2.13.1/ $ apisix start                                                                                               [10:32:24]
/usr/local/Cellar/openresty/1.19.9.1_2/luajit/bin/luajit ./apisix/cli/apisix.lua start
got malformed key-put message: "{"error":"put is not supported","code":2,"message":"put is not supported"}" from etcd "http://127.0.0.1:12379/v3/kv/put"
```

**Comment by membphis** [comment]
@soulbird pls take a look at this issue, many thx

**Comment by bzp2010** [comment]
@siaron Yes, the current `Kine` dependency does not support `put` and `delete` functions; there is currently a PR implementation of this feature #4 , but it has been implemented in a very **hack** way, and we plan to refactor this feature, but currently lack sufficient manpower to complete this work.

**Comment by siaron** [comment]
Happy to be a part of it.  Are there any relevant design documents or materials?

**Comment by ZhangShangyu** [comment]
so this etcd-adapter can not work as apisix etcd now?

**Comment by tokers** [comment]
> so this etcd-adapter can not work as apisix etcd now?

It can do a lot but it's still an experimental project. You try play it but be careful when you want to use it on prod env.

**Comment by ZhangShangyu** [comment]
> 

thanks. let me try try
